### ISSUE TITLE
Retrieve VOIP token

### ISSUE THREAD

**Issue by marcoberetta96** [hiennguyen92/flutter_callkit_incoming#19]
I was wondering about how to retrieve the VOIP token (from flutter) to push it to a custom server.
It seems it's currently not possible.

I found this package https://pub.dev/packages/flutter_ios_voip_kit which allows to retrieve the token.
This package is very old and not maintained anymore.
It would be nice to have that feature here.

**Comment by zionnite** [comment]
Please where you able to make use of VoIP on your app?
If yes? Can you please teach me ?🥺

I don't even know what to do with VoIP, it's confusing

**Comment by marcoberetta96** [comment]
I have not a completely working application. I'm waiting for a PR here and for a pair of issues.

For now you can retrieve the VoIP token using this code.
https://stackoverflow.com/questions/37099280/didupdatepushcredentials-not-get-called

Then you can try to send a VoIP notification using for example OneSignal and see what happens :)

**Comment by zionnite** [comment]
thanks man @marcoberetta96

**Comment by hiennguyen92** [comment]
Hi @marcoberetta96 @zionnite

I just updated version 1.0.1
please update it

also i have a tutorial on setup Pbushkit 
https://github.com/hiennguyen92/flutter_callkit_incoming/blob/master/PUSHKIT.md

**Comment by marcoberetta96** [comment]
Hi @hiennguyen92 and many thanks for your work!
I will try and test all the new stuff in some hours.

Just to be sure... when this package retrieves the VOIP token, would it be possible to pass it from native code to dart code? It would be much simpler to send it to a custom server by using dart. Thanks again!

**Comment by hiennguyen92** [comment]
Hi @marcoberetta96 
when you receive Push VoIP, the app will wake up automatically. 

also make sure the call listener is at the root of the App
FlutterCallkitIncoming.onEvent.listen...
Use it to get listener Incoming, Accept, Decline event

**Comment by hiennguyen92** [comment]
@marcoberetta96 
for send device token from native to dart, I think there may still be something missing, I think I will update you soon.

**Comment by marcoberetta96** [comment]
@hiennguyen92 
Got it.
I think that sending the device token from native to dart is a must-have feature for this package.
Anyway, you already do nice things, thank you!

**Comment by hiennguyen92** [comment]
Hi @marcoberetta96 

I just updated version 1.0.1+1

using getDevicePushTokenVoIP for get device token VoIP

```dart
await FlutterCallkitIncoming.getDevicePushTokenVoIP();
```

please make sure set token from native
```swift
func pushRegistry(_ registry: PKPushRegistry, didUpdate credentials: PKPushCredentials, for type: PKPushType) {
    print(credentials.token)
    let deviceToken = credentials.token.map { String(format: "%02x", $0) }.joined()
    //Save deviceToken to your server
    SwiftFlutterCallkitIncomingPlugin.sharedInstance?.setDevicePushTokenVoIP(deviceToken)
}

func pushRegistry(_ registry: PKPushRegistry, didInvalidatePushTokenFor type: PKPushType) {
    print("didInvalidatePushTokenFor")
    SwiftFlutterCallkitIncomingPlugin.sharedInstance?.setDevicePushTokenVoIP("")
}
```

**Comment by marcoberetta96** [comment]
Nice work, thanks!
Much appreciated :)

**Comment by zionnite** [comment]
> Hi @marcoberetta96 when you receive Push VoIP, the app will wake up automatically.
> 
> also make sure the call listener is at the root of the App FlutterCallkitIncoming.onEvent.listen... Use it to get listener Incoming, Accept, Decline event

making a call at the root of the app will be in the main.dart, right?

but my coding its calling from a different file, how do u suggest i do this?

**Comment by hiennguyen92** [comment]
@zionnite 

just be sure to call it in App stateful.
Once you receive the event, call whatever function you want in your app

**Comment by zionnite** [comment]
thanks so much...
would give you feedback.

thanks again

**Comment by zionnite** [comment]
Hello sir

i keep getting the error

My podfile is pointing to   `platform :ios, '12.0'`

My AppFrameworkInfo.plist is `minimum requirement is pointing to 12.0`

```
Swift Compiler Error (Xcode): Compiling for iOS 9.0, but module 'flutter_callkit_incoming' has a minimum deployment target of iOS 12.0: /Users/apple/StudioProjects/prayer_buddy/build/ios/Debug-iphonesimulator/flutter_callkit_incoming/flutter_callkit_incoming.framework/Modules/flutter_callkit_incoming.swiftmodule/x86_64-apple-ios-simulator.swiftmodule
/Users/apple/StudioProjects/prayer_buddy/ios/Runner/AppDelegate.swift:3:7

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
```


i started getting the error once i updated my `appDelegate.swift` file

**Comment by zionnite** [comment]
i create another flutter application after setting up my configuration with this package, i got the following again

```
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Running pod install...
Running Xcode build...
Xcode build done.                                           110.6s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/yy/s60fqgn15hb_gql22vlb2z540000gn/T/flutter_tools.zjuIF4/flutter_ios_build_temp_dirIaBt7R/temporary_xcresult_bundle

    /Users/apple/IdeaProjects/brayer_pals/ios/Pods/Reachability/Reachability.m:465:63: warning: cast to smaller integer type 'unsigned int' from 'Reachability *' [-Wpointer-to-int-cast]
                                 NSStringFromClass([self class]), (unsigned int) self, [self currentReachabilityFlags]];
                                                                  ^~~~~~~~~~~~~~~~~~~
    1 warning generated.
    /Users/apple/.pub-cache/hosted/pub.dartlang.org/flutter_callkit_incoming-1.0.1+6/ios/Classes/SwiftFlutterCallkitIncomingPlugin.swift:269:12: warning: unnecessary check for 'iOS'; enclosing scope ensures guard will always be true
            if #available(iOS 11.0, *) {
               ^
    /Users/apple/.pub-cache/hosted/pub.dartlang.org/flutter_callkit_incoming-1.0.1+6/ios/Classes/SwiftFlutterCallkitIncomingPlugin.swift:7:14: note: enclosing scope here
    public class SwiftFlutterCallkitIncomingPlugin: NSObject, FlutterPlugin, CXProviderDelegate {
                 ^
    /Users/apple/.pub-cache/hosted/pub.dartlang.org/flutter_callkit_incoming-1.0.1+6/ios/Classes/SwiftFlutterCallkitIncomingPlugin.swift:189:34: warning: expression implicitly coerced from '[String : Any?]' to '[String : Any]'
            return self.callManager?.activeCalls()
                                     ^~~~~~~~~~~
    /Users/apple/.pub-cache/hosted/pub.dartlang.org/flutter_callkit_incoming-1.0.1+6/ios/Classes/SwiftFlutterCallkitIncomingPlugin.swift:189:34: note: explicitly cast to '[String : Any]' with 'as [String : Any]' to silence this warning
            return self.callManager?.activeCalls()
                                     ^~~~~~~~~~~
                                                 as [String : Any]
    /Users/apple/IdeaProjects/brayer_pals/ios/Runner/AppDelegate.swift:4:8: error: compiling for iOS 9.0, but module 'flutter_callkit_incoming' has a minimum deployment target of iOS 11.0: /Users/apple/IdeaProjects/brayer_pals/build/ios/Debug-iphonesimulator/flutter_callkit_incoming/flutter_callkit_incoming.framework/Modules/flutter_callkit_incoming.swiftmodule/x86_64-apple-ios-simulator.swiftmodule
    import flutter_callkit_incoming
           ^
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

    Result bundle written to path:
    	/var/folders/yy/s60fqgn15hb_gql22vlb2z540000gn/T/flutter_tools.zjuIF4/flutter_ios_build_temp_dirIaBt7R/temporary_xcresult_bundle


Swift Compiler Error (Xcode): Compiling for iOS 9.0, but module 'flutter_callkit_incoming' has a minimum deployment target of iOS 11.0: /Users/apple/IdeaProjects/brayer_pals/build/ios/Debug-iphonesimulator/flutter_callkit_incoming/flutter_callkit_incoming.framework/Modules/flutter_callkit_incoming.swiftmodule/x86_64-apple-ios-simulator.swiftmodule
/Users/apple/IdeaProjects/brayer_pals/ios/Runner/AppDelegate.swift:3:7

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
```

**Comment by zionnite** [comment]
Update
when i roll back my ` AppDelegate.swift` code to the flutter generated code, everything works fine for me


```
import UIKit
import Flutter

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    GeneratedPluginRegistrant.register(with: self)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}
```




**code run without an error**

how can this be resolved???

**Comment by zionnite** [comment]
![Screen Shot 2022-04-09 at 5 11 19 PM](https://user-images.githubusercontent.com/10984801/162582231-0360c610-67f2-4408-80d0-4a7792662a36.png)
![Screen Shot 2022-04-09 at 5 12 49 PM](https://user-images.githubusercontent.com/10984801/162582234-023b0e46-b83d-4cd0-b030-4a010706f498.png)

**Comment by zionnite** [comment]
@hiennguyen92 sir can you please quickly look into this, have done everything possible to make this thing work out but its not working for me at all

**Comment by hiennguyen92** [comment]
<img width="1129" alt="image" src="https://user-images.githubusercontent.com/3755347/162633686-efbd5426-3fc8-41c7-86c2-8adc1cb900d3.png">
i think you can try change 11 -> 12

**Comment by zionnite** [comment]
I have Tired 11,12,13 and 14 yet the error still persist

**Comment by zionnite** [comment]
> I have Tired 11,12,13 and 14 yet the error still persist

It's was on 12 before I changed it to 11

**Comment by zionnite** [comment]
just created a new flutter app and added this package to it and the `**issue still persist**`

am waiting on you sir, i need your assistance on this

**Comment by zionnite** [comment]
@hiennguyen92 Please I need your support
